### ISSUE TITLE
Entity mutators documentation misleading

### ISSUE THREAD

**Issue by gerhard-lechner** [cakephp/docs#7330]
I think there's an error in the docs when it comes to explaining Entity mutators. At https://book.cakephp.org/4/en/orm/entities.html#Cake\ORM\Entity::set it says:

> Mutator methods should always return the value that should be stored in the field. **_As you can see above_**, you can also use mutators to set other calculated fields. When doing this, be careful to not introduce any loops, as CakePHP will not prevent infinitely looping mutator methods.

I think the example mentioned here is actually missing? And then further down below there's this code:

```
$user->title = 'foo'; // slug is set as well
$user->set('title', 'foo'); // slug is set as well
```

Looking at the `_setTitle()` method just above this code, I think the comments here are misleading. The slug isn't set but the `title` field is converted to a slug.

Overall I think this section could look something like this to make more sense:
____________________________

You can customize how fields get set by defining a mutator. Mutators use the convention of `_set` followed by the CamelCased version of the field name.

**Cake\ORM\Entity::set($field = null, $value = null)**

Mutators allow you to convert fields as they are set. They should always return the value that should be stored in the field. For example:

```
namespace App\Model\Entity;

use Cake\ORM\Entity;
use Cake\Utility\Text;

class Article extends Entity
{
    protected function _setTitle($title)
    {
        return ucwords($title);
    }
}
```

You can also use mutators to set other calculated fields. When doing this, be careful to not introduce any loops, as CakePHP will not prevent infinitely looping mutator methods.

```
namespace App\Model\Entity;

use Cake\ORM\Entity;
use Cake\Utility\Text;

class Article extends Entity
{
    protected function _setTitle($title)
    {
        $this->slug = Text::slug($title);

        return ucwords($title);
    }
}
```

(then continue with "The mutator would be run ...")
____________________________

Of course it can also be shortened to only have the last code example which basically shows conversion of the value as well as setting a calculated field. However, I think the way it is explained right now makes it hard to understand what's actually going on. What do you guys think?

**Comment by markstory** [comment]
Those changes seem reasonable to me.

**Comment by othercorey** [comment]
> You can customize how fields get set by defining a mutator. Mutators use the convention of _set followed by the CamelCased version of the field name.

I would use `_set(FieldName)` followed by the explanation of camel casing the field name. It should be easier for some users to read the english.

**Comment by gerhard-lechner** [comment]
This sentence is actually a copy of what's written in the previous section about Accessors.

> Accessors use the convention of `_get` followed by the CamelCased version of the field name.

So it would make sense to change this in both places then. I might just create a PR updating the "Accessors & Mutators" section according to the things discussed here.

**Comment by gerhard-lechner** [comment]
Solved with [c98e8b6](https://github.com/cakephp/docs/commit/c98e8b6a8f1da14a7cc9ada1257605cc3eae5ac6)
